trio/_wait_for_object.py

```diff
@@ -30,24 +34,36 @@ async def WaitForSingleObject(obj):
     if retcode == ErrorCodes.WAIT_FAILED:
         raise_winerror()
     elif retcode != ErrorCodes.WAIT_TIMEOUT:
+        await trio.lowlevel.cancel_shielded_checkpoint()
```

Comment:
I think we can make this a plain `trio.lowlevel.checkpoint()` and remove the `checkpoint_if_cancelled` at the top of the function?

Reply:
I thought `checkpoint_if_cancelled` was a nice micro-optimization for avoiding system calls if this function is already cancelled before entering, but a plain checkpoint is easier to understand for sure.